File: android-native-shell-scaffold/app/src/main/java/com/vinesight/shell/NativeRouteRegistry.kt
package com.vinesight.shell

enum class NativeRouteId(val raw: String) {
    AUTH_LOGIN("auth.login"),
    AUTH_PHONE_LOGIN("auth.phone_login"),
    AUTH_OTP_VERIFICATION("auth.otp_verification"),
    AUTH_PROFILE_COMPLETION("auth.profile_completion"),
    ONBOARDING("onboarding"),
    TABS_HOME("tabs.home"),
    TABS_FARMS("tabs.farms"),
    TABS_TOOLS("tabs.tools"),
    TABS_WORKERS("tabs.workers"),
    TABS_SETTINGS("tabs.settings")
}

enum class NativeTabId {
    HOME,
    FARMS,
    TOOLS,
    WORKERS,
    SETTINGS
}

object NativeShellRegistry {
    val tabs: List<NativeTabId> = NativeTabId.entries
    val routes: List<NativeRouteId> = NativeRouteId.entries
}

File: ios-native-shell-scaffold/VineSightShell/NativeRouteRegistry.swift
import Foundation

enum NativeRouteId: String, CaseIterable {
    case authLogin = "auth.login"
    case authPhoneLogin = "auth.phone_login"
    case authOtpVerification = "auth.otp_verification"
    case authProfileCompletion = "auth.profile_completion"
    case onboarding = "onboarding"
    case tabsHome = "tabs.home"
    case tabsFarms = "tabs.farms"
    case tabsTools = "tabs.tools"
    case tabsWorkers = "tabs.workers"
    case tabsSettings = "tabs.settings"
}

enum NativeTabId: String, CaseIterable {
    case home
    case farms
    case tools
    case workers
    case settings
}

struct NativeShellRegistry {
    static let tabs: [NativeTabId] = NativeTabId.allCases
    static let routes: [NativeRouteId] = NativeRouteId.allCases
}
